blocks/library/image/block.js

```diff
@@ -293,12 +293,17 @@ class ImageBlock extends Component {
 			</InspectorControls>
 		);
 
+		// Pass optional width as variable
+		const figureWidth = {
+			'--caption-width': width + 'px',
+		};
+
 		// Disable reason: Each block can be selected by clicking on it
 		/* eslint-disable jsx-a11y/no-static-element-interactions, jsx-a11y/onclick-has-role, jsx-a11y/click-events-have-key-events */
 		return (
 			<Fragment>
 				{ controls }
-				<figure className={ classes }>
+				<figure className={ classes } style={ figureWidth }>
```

Comment:
This only fixes the issue on the editor, not on the frontend. Also, why can't we apply the width directly to the figure once resized instead of using css variables?

Reply:
> This only fixes the issue on the editor, not on the frontend. 

That was a result of this being a "try" branch. I can very easily port it to the front-end if we agree that this is the way to go, which brings me to your next very good question:

> why can't we apply the width directly to the figure once resized instead of using css variables?

The problem with inline styles is that themes can't override them without using `!important`. So if a theme wanted to do something like has been done in https://codepen.io/joen/pen/oEYVXB, which makes the `figure` be the same width as the main column, and then resizes the individual `img` and `figcaption` inside that figure (to enable wide layouts with floats), then they'd have to use `!important` to override the inline style.

By using a CSS variable, we enable themers to be able to do what they want — if they want to do nothing, the CSS we provide should size things properly. If they want to override it, they can do that, without having to use `!important`. 

I know it's somewhat exotic CSS, but it feels like it walks the line and brings us the best of both worlds. What do you think?

Also CC @pento as you tried to introduce `fit-content` a while back. 

LMK and I'll add this to the frontend.

Reply:
Thanks for the clarification, the problem is the css variables don't work on IE11. Maybe this could be considered an "ok" regression for IE11, I don't really know?!

Reply:
Good questions. For a graceful IE11 semi-support, we could do something like this:

```
figcaption {
	max-width: 50%;
	max-width: var( --caption-width ); // use optional width variable for captions
}
```

The max-width would then be overridden by the CSS variable, when supported, but fall back to _half_ the width of the main column when not. 

It's still hack territory, but I still think it's worth it, but as always will defer to others opinions if strongly held.

I ran into an issue which probably needs @pento input. It seems I can't output `--caption-width: ...` on the front-end. I think wptexturize eats the leading hyphens, in any case when output I just see `caption-width: ...` in the rendered code. Are there any tricks to getting this to work, or is that what'll block this?